src/main/java/bio/terra/service/dataset/DatasetJsonConversion.java

```diff
@@ -100,6 +118,18 @@ public static GoogleRegion getRegionFromDatasetRequestModel(DatasetRequestModel
         }).collect(Collectors.toList());
     }
 
+    private static List<? extends StorageResource<?, ?>> createAzureStorageResourceValues(
+        DatasetRequestModel datasetRequestModel) {
+        final AzureRegion region = getAzureRegionFromDatasetRequestModel(datasetRequestModel);
+        // TODO: once we no longer require GCP resources to back Azure datasets, stop concatenating
+        return Stream.of(
+            Arrays.stream(AzureCloudResource.values()).map(resource -> new AzureStorageResource(null,
+                resource,
+                region)),
+            createGcpStorageResourceValues(datasetRequestModel).stream()).flatMap(s -> s)
```

Comment:
Instead of `flatMap()` you can directly concatenate the two streams.
```suggestion
        return Stream.concat(
            Arrays.stream(AzureCloudResource.values()).map(resource -> new AzureStorageResource(null,
                resource,
                region)),
            createGcpStorageResourceValues(datasetRequestModel).stream())
```